src/adminEntity.ts

```diff
@@ -91,6 +90,37 @@ abstract class AdminEntity {
     validateFieldsExist(this, 'searchFields', this.metadata)
     validateFieldsAreNotRelation(this, 'searchFields', this.metadata)
   }
+
+  buildSearchQueryOptions(
+    query: SelectQueryBuilder<unknown>, // @debt typing "miker: can I type this better?"
+    options: { alias: string; searchParam: string },
```

Comment:
```suggestion
    alias: string, searchParam: string
```

Reply:
why is this better?

Reply:
This "options" thing doesn't make semantic sense to me, what does it represent? Does it provide an abstraction, or is it just a bunch of parameters necessary for the function execution?